homeassistant/config_entries.py

```diff
@@ -1169,6 +1175,52 @@ def _data_to_save(self) -> dict[str, list[dict[str, Any]]]:
         """Return data to save."""
         return {"entries": [entry.as_dict() for entry in self._entries.values()]}
 
+    async def async_remove_device(self, entry_id: str, device_id: str) -> bool:
+        """Remove a device."""
+        if (config_entry := self.async_get_entry(entry_id)) is None:
+            return False
```

Comment:
Should we raise specific errors for each problem? Maybe a `HomeAssistantError` with a description?

Reply:
Yes, raising errors sounds good. I would raise specific errors for different cases. Maybe `MissingEntry` for missing device entry or config entry id?

Reply:
Updated in 4baf5ef